File: Lab2Java/app/src/main/java/com/mario21ic/MainActivity.java
package com.mario21ic;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.content.pm.ActivityInfo;
import android.os.Bundle;
import android.view.Gravity;
import android.view.View;

import android.widget.Toast;
import android.view.Menu;
import android.view.MenuItem;
import android.util.Log;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);
        setContentView(R.layout.activity_main);

        // Llamar a Principal
//        Handler handler = new Handler();
//        handler.postDelayed(new Runnable() {
//            @Override
//            public void run() {
//                Intent intent = new Intent(MainActivity.this, PrincipalActivity.class);
//                startActivity(intent);
//            }
//        }, 2000);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.principal, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle item selection
        switch (item.getItemId()) {
            case R.id.menu_inicio:
                Log.i("====>","Click en Inicio!!");
                startActivity(new Intent(this, MainActivity.class));
                return true;
            case R.id.menu_productos:
                Log.i("====>","Click en Productos!!");
                startActivity(new Intent(this, ProductosActivity.class));
                return true;
            case R.id.menu_servicios:
                Log.i("====>","Click en Servicios!!");
                startActivity(new Intent(this, ScrollActivity.class));
                return true;
            default:
                return super.onOptionsItemSelected(item);
        }
    }

    public void btnClickeame(View view) {
        Toast t = Toast.makeText(getApplicationContext(), "Acabas de presionar Clickeame", Toast.LENGTH_LONG);
        t.setGravity(Gravity.BOTTOM | Gravity.RIGHT, 0,0);
        t.show();
    }

    public void btnProductos(View view) {
        setContentView(R.layout.activity_productos);
    }

    public void btnScroll(View view) {
        setContentView(R.layout.activity_scroll_activity);
    }

}

File: Lab2Kotlin/app/src/main/java/com/mario21ic/MainActivity.kt
package com.mario21ic

import android.content.Intent
import android.content.pm.ActivityInfo
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.Menu
import android.view.MenuItem

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
        setContentView(R.layout.activity_main)
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.principal, menu)
        return super.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.menu_inicio -> {
                Log.i("===", "Click de Inicio!")
                return true
            }
            R.id.menu_productos -> {
                Log.i("===", "Click de Productos!")
                val intent = Intent(this, ProductosActivity::class.java)
                startActivity(intent)
                return true
            }
            R.id.menu_servicios -> {
                Log.i("===", "Click de Servicios!")
                return true
            }
        }
        return super.onOptionsItemSelected(item)
    }
}